zeus/run/dataloader.py

```diff
@@ -1078,8 +1091,13 @@ def __next__(self):
                     f"Profile iterations {self.sample_num - self.prof_start_sample}"
                     f" exceeds `self.profile_iter` {self.profile_iter}."
                 )
-            # We weren't doing anything. Start warming up.
-            if self.prof_state == NOT_PROFILING:
+            # We weren't doing anything. Start warming up if the iterations left in
+            # the current epoch can accommodate at least one profile window.
+            if (
+                self.prof_state == NOT_PROFILING
+                and self.sample_num + self.warmup_iter + self.profile_iter
+                < self.num_samples
+            ):
```

Comment:
Not 100% sure, but given we have this, do we need the sanity checks in lines 1084 - 1093?